diff --git a/am335x-boneblack.dts.orig b/am335x-boneblack.dts
index 09ea639..58dc61e 100644
--- a/arch/arm/boot/dts/ti/omap/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/ti/omap/am335x-boneblack.dts
@@ -178,3 +178,13 @@
 &baseboard_eeprom {
 	vcc-supply = <&ldo4_reg>;
 };
+
+&i2c2 {
+    status = "okay";
+    clock-frequency = <100000>;
+    bno055@28 {
+		status = "okay";
+		compatible = "bosch,bno055";
+		reg = <0x28>;
+	};
+};
