diff --git a/am335x-boneblack.dts.orig b/am335x-boneblack.dts
index 09ea639..5a02bb8 100644
--- a/arch/arm/boot/dts/ti/omap/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/ti/omap/am335x-boneblack.dts
@@ -178,3 +178,136 @@
 &baseboard_eeprom {
 	vcc-supply = <&ldo4_reg>;
 };
+
+/* =========================
+ * I2C2 Devices (FIX PINMUX)
+ * ========================= */
+&i2c2 {
+	status = "okay";
+	clock-frequency = <400000>;
+
+    bno055@28 {
+        compatible = "desmtiny,bno055";
+        reg = <0x28>;
+    };
+
+    ds3231rtc@68 {
+        compatible = "desmtiny,ds3231rtc";
+        reg = <0x68>;
+    };
+
+    bmp280@76 {
+        compatible = "desmtiny,bmp280";
+        reg = <0x76>;
+    };
+
+    at24c256@50 {
+        compatible = "atmel,24c256";
+        reg = <0x50>;
+        pagesize = <64>;
+    };
+};
+
+
+/* =========================
+ * Pinmux
+ * ========================= */
+&am33xx_pinmux {
+    /* LED */
+    user_led_pins: pinmux_user_led_pins {
+        pinctrl-single,pins = <
+            0x030 0x07
+        >;
+    };
+
+    /* BUZZER */
+    buzzer_pins: pinmux_buzzer_pins {
+        pinctrl-single,pins = <
+            0x034 0x07
+        >;
+    };
+
+    /* PWM ENABLE */
+    pwmena1a_pins: pinmux_pwmena1a_pins {
+        pinctrl-single,pins = <0x03c 0x07>;
+    };
+
+    pwmena2a_pins: pinmux_pwmena2a_pins {
+        pinctrl-single,pins = <0x038 0x07>;
+    };
+
+    pwmena1b_pins: pinmux_pwmena1b_pins {
+        pinctrl-single,pins = <0x02c 0x07>;
+    };
+
+    pwmena2b_pins: pinmux_pwmena2b_pins {
+        pinctrl-single,pins = <0x028 0x07>;
+    };
+
+
+    /* PWM */
+    pwm1_pins: pinmux_pwm1_pins {
+        pinctrl-single,pins = <
+            0x048 0x06
+        >;
+    };
+};
+
+/* =========================
+ * PWM
+ * ========================= */
+&epwmss1 {
+    status = "okay";
+};
+
+&ehrpwm1 {
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&pwm1_pins>;
+};
+
+
+/* =========================
+ * GPIO Devices
+ * ========================= */
+&ocp {
+
+    user_leds {
+        compatible = "gpio-leds";
+        pinctrl-names = "default";
+        pinctrl-0 = <&user_led_pins>;
+
+        led0 {
+            label = "user-led";
+            gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+        };
+    };
+
+    buzzer {
+        compatible = "gpio-leds";
+        pinctrl-names = "default";
+        pinctrl-0 = <&buzzer_pins>;
+
+        buzzer0 {
+            label = "buzzer";
+            gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+        };
+    };
+
+    pwm_enable {
+        compatible = "gpio-leds";
+
+        pinctrl-names = "default";
+        pinctrl-0 = <
+            &pwmena1a_pins
+            &pwmena2a_pins
+            &pwmena1b_pins
+            &pwmena2b_pins
+        >;
+
+        pwmena1a { gpios = <&gpio1 15 GPIO_ACTIVE_HIGH>; };
+        pwmena2a { gpios = <&gpio1 14 GPIO_ACTIVE_HIGH>; };
+        pwmena1b { gpios = <&gpio0 27 GPIO_ACTIVE_HIGH>; };
+        pwmena2b { gpios = <&gpio2 1 GPIO_ACTIVE_HIGH>; };
+    };
+};
\ No newline at end of file
